src/libs/antares/study/parts/hydro/pair-of-integers.h

```diff
@@ -0,0 +1,17 @@
+#pragma once
+
+class PairOfIntegers
+{
+public:
+    PairOfIntegers(unsigned int numberOfTS_1, unsigned int numberOfTS_2);
+
+    bool same() const;
+    bool bothGreaterThanOne() const;
+    unsigned int sup() const { return numberOfTSsup_; }
+
+private:
+    unsigned int numberOfTS_1_;
+    unsigned int numberOfTS_2_;
+    unsigned int numberOfTSsup_{ 0 };
+    unsigned int numberOfTSinf_{ 0 };
+};
```

Comment:
Not sure it would be useful, and here is why (in my opinion) :
- Let's take an exemple from another field : sort algorithm. From a comparison between 2 elements, sort algorithm manages to sort the elements of a list.
Here, the idea would be the same : use the comparison between 2 elements to compare many of them.
So we should keep this class as it is, and build on it.
- As you said, we have **ror**, **storage** and **mingen** time series, and we have to "equalize" these sizes. It's done in function **DataSeriesHydro::EqualizeGenerationTSsizes**. As you can see in this function's body, there is some code duplication : we have 2 blocks very much alike. This is usually a signe that we should refactor to remove the code duplication. I spent some time to get to this results on reading the hydro TS, and there's still things to do, apart from removing this code duplication which is, in fact the subject you're talking about here.
To me, we should first try to merge this as it is, create a gitHub issue if we're not happy, and fic this later.